streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -782,7 +849,12 @@ private KafkaStreams(final InternalTopologyBuilder internalTopologyBuilder,
                 cacheSizePerThread,
                 stateDirectory,
                 delegatingStateRestoreListener,
-                i + 1);
+                i + 1,
+                KafkaStreams.this::close,
```

Comment:
This will call closeToError but I am testing if that has a problem. So far it does not